### PR TITLE
#105 pdd should store github user name not real name

### PR DESCRIPTION
**Pull request by a6cexz**
Use github search api to obtain github login using author email

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#107](https://github.com/yegor256/pdd/issues/107) is now in scope, role is `REV`

File: lib/pdd/source.rb [hunk 1]
```diff
@@ -151,11 +153,47 @@ def git(pos)
...
+      login = find_github_login(info[:email])
+      if login.length > 0
+        info[:author] = login
+      end
+      info
+    end
+
+    def get_json(query)
+      uri = URI.parse(query)
+
```
Comment (yegor256) [inline lib/pdd/source.rb]: @a6cexz see https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

**Comment by yegor256** [comment]
@a6cexz how do you know that this code works?

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#107](https://github.com/yegor256/pdd/issues/107); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by a6cexz** [comment]
@yegor256 I've added test case for getting github login from author email

File: lib/pdd/source.rb (hunk 1 above)
Comment (a6cexz) [reply lib/pdd/source.rb]: Thanks, corrected

**Comment by yegor256** [comment]
@a6cexz get rid of empty lines inside method bodies: https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

Aside from that, I think this functionality should be backward compatible. I mean, if we can find the GitHub account name, we set author to `@yegor256`. If we can't find it, we leave it as `Yegor Bugayenko`. In other words, if we find it, we start with `@`.

**Comment by a6cexz** [comment]
@yegor256 done

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/pdd#107` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#107](https://github.com/yegor256/pdd/issues/107) is now out of scope

**Pull request merged**